pkg/kubelet/cm/devicemanager/manager.go

```diff
@@ -1013,6 +1025,12 @@ func (m *ManagerImpl) GetDeviceRunContainerOptions(pod *v1.Pod, container *v1.Co
 		if err != nil {
 			return nil, err
 		}
+
+		if !m.checkPodActive(pod) {
+			klog.ErrorS(nil, "pod deleted from activePods, skip to reAllocate", "podUID", podUID)
+			continue
+		}
+
```

Comment:
I don't believe we need this check at all. I think we just need to remove the entire code block below:
```
		// This is a device plugin resource yet we don't have cached
		// resource state. This is likely due to a race during node
		// restart. We re-issue allocate request to cover this race.
		if m.podDevices.containerDevices(podUID, contName, resource) == nil {
			needsReAllocate = true
		}
```
As well as this one:
```
	if needsReAllocate {
		klog.V(2).InfoS("Needs to re-allocate device plugin resources for pod", "pod", klog.KObj(pod), "containerName", container.Name)
		if err := m.Allocate(pod, container); err != nil {
			return nil, err
		}
	}
```
Without any additional checks or errors being propagated back up from this function.

Reply:
@klueska the reason why I keep this logic is that I can only reproduce the panic by steps showing above, but I can't tell for sure whether there is another scenario that we may need to re-allocate devices. in order to make sure any pod requesting for devices won't be running without them (which may cause a unexpected error for online services), we prefer to panic in our production environment, so I add this check rather than just remove them all. but if you insist, please make me know, I can try to delete . thx

Reply:
Yes. Please remove it as suggested.

As I mentioned before, now that we do allocation at pod admit time (rather than during container start) calling `Allocate()` in this function is actually incorrect.

Specifically, the changes in 0d68bffd03f1af15fc8cbb3eaeda9ff95a47eeab moved allocation from the container start path to the pod admit path, at which point this code in `GetDeviceRunContainerOptions()` should have been removed (but was overlooked).

Reply:
Actually thinking about this a bit more, let's go with your proposal (to at least mask the underlying issue) and then reevaluate if it makes sense to make the change I propose above at a later time.